src/main/java/com/avairebot/commands/fun/PunCommand.java

```diff
@@ -0,0 +1,187 @@
+package com.avairebot.commands.fun;
+
+import com.avairebot.AvaIre;
+import com.avairebot.chat.SimplePaginator;
+import com.avairebot.commands.CommandMessage;
+import com.avairebot.contracts.commands.Command;
+import com.avairebot.contracts.commands.CommandContext;
+import com.avairebot.factories.RequestFactory;
+import com.avairebot.language.I18n;
+import com.avairebot.requests.Response;
+import com.avairebot.requests.service.PunService;
+import com.avairebot.utilities.NumberUtil;
+import org.json.JSONArray;
+import org.json.JSONObject;
+import org.jsoup.helper.StringUtil;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.function.Consumer;
+
+public class PunCommand extends Command
+{
+    private final String templateUrl = "https://icanhazdadjoke.com/";
+
+    public PunCommand(AvaIre avaire)
+    {
+        super(avaire);
+    }
+
+
+
+    /**
+     * Gets the command name, this is used in help and error
+     * messages as the title as well as log messages.
+     *
+     * @return Never-null, the command name.
+     */
+    @Override
+    public String getName() {
+        return "Pun Command";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Gets a random pun or a list of puns based on a search query.";
+    }
+
+
+    @Override
+    public List<String> getUsageInstructions() {
+        return Arrays.asList(
+            "`:command` - Gets a random pun",
+                "`:command` <query> <page> - retrieves a list of puns ");
```

Comment:
The `<query> <page>` should both be under the `` ` `` character to mark the whole command instead of just the trigger, and the page should also be using square brackets(`[]`) to mark it as an optional argument.

Reply:
Made it under the tilde and the page is now in brackets.